src/index.ts

```diff
@@ -7,18 +7,48 @@ import {
 } from './config'
 import { fromQuery, fromUntilQuery, untilQuery } from './media-queries'
 
-type Until = { [key in Breakpoints]: string }
+/*
+`from` and `until` are objects with overridden `toString` methods,
+so that they can be coerced into media query strings _or_ provide
+further refinement of the query. this freaky genius was thought up
+by @SiAdcock.
+
+e.g.:
+
+const styles = {
+	[from.small]: {
+		color: 'red'
+	},
+	[from.small.until.large]: {
+		color: 'blue'
+	},
+	[from.small.until.large.for('print')]: {
+		color: 'black'
+	},
+}
+*/
+
+type Until = {
+	[key in Breakpoints]: {
+		toString: () => string
+		for: (arg0: string) => string
+	}
+}
 
 type From = {
 	[key in Breakpoints]: {
 		toString: () => string
 		until: Until
+		for: (arg0: string) => string
 	}
 }
 
 export const until = Object.entries(breakpoints).reduce(
 	(untils, [untilName, untilWidth]) => ({
-		[untilName]: untilQuery(untilWidth),
+		[untilName]: {
+			toString: () => untilQuery(untilWidth),
+			for: (type: string) => untilQuery(untilWidth, type),
```

Comment:
Seriously now, I think the media type is a list of `or` conditions, so would be better suited as an array

`for(['screen', 'print'])` => `@media screen, print`

I say this because `@media screen and print` doesn't fit the spec. A device won't be both screen _and_ print, it's one or the other.

Reply:
this is a very good point, seems i don't really know my around media queries that well!

what do you think of `for('screen', 'print', etc)` rather than an array? it feels nicer to me, but not sure if it's a bit 'magic'?

Reply:
Yeah that’s a nice idea. It takes advantage of language features, so it doesn’t feel magic to me

Reply:
are we on the right track here?

let say you did

```js
const styles = {
	[from.small.for('screen', 'print')]: {
		color: 'red'
	}
}

``` 

you'd get:

```css
@media screen, print and (min-width: xxx) { color: red }
```

as i understand it, this is equivalent to 

```scss
@media screen { y }
// OR 
@media print and (min-width: xxx) { color: red }
```
?

should we output this in instead?

```css
@media screen and (min-width: xxx), print and (min-width: xxx) { color: red }
```

Reply:
perhaps too we should log something in dev mode if your `for` args contain a comma?

Reply:
> should we output this in instead?
> `@media screen and (min-width: xxx), print and (min-width: xxx) { color: red }`

Goodness me, you are right 😲I had not considered how much complexity such a small feature would bring!

👍to debug logging, especially if we can produce separate builds for development and production, and strip out the logging in production? Maybe that's a later optimisation!

Reply:
The limitation with this solution is that we can't do something like this in one statement:

```css
@media screen and (min-width: 30em), 
       print  and (min-width: 40em) { 
  color: red 
}
```

But maybe this is fine!

Reply:
i wonder if we're worrying too much? sass-mq just uses whatever you pass in, and i'm not aware it's caused any problems?

if you want a really complex query, this is probably not a useful lib, and maybe we should be ok with that? until it's a problem anyway?

Reply:
Yeah you’re right, I think I got carried away after reading the readme PR! Happy for us to keep it like this for now